### PR TITLE
JBPM-8067: Magnets & connector bendpoints are not working with the scrollable lienzo panel.

### PR DESCRIPTION
**Pull request by romartin**
Hey @tiagodolphine  @hasys 

Here are the tests for https://github.com/kiegroup/lienzo-core/pull/79

Thanks!

### CODE REVIEW

**Comment by hasys** [comment]
Hi @romartin looks, good but I am still not able to:
* Reconnect bendpoint from one node to another if nodes placed out of the canvas. I am able to disconnect the point from the node, but not able to connect it again/reconnect to another point.
* Add new bendpoint if sequence flow is out of the canvas. New nice animation shows successfully, but new bendpoint is not created.

Will check the code now.

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlImplTest.java [hunk 1]
```diff
@@ -62,20 +41,10 @@
...
-import static org.mockito.Mockito.any;
-import static org.mockito.Mockito.anyDouble;
-import static org.mockito.Mockito.atLeastOnce;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.never;
-import static org.mockito.Mockito.spy;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
```
Comment (hasys) [inline src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlImplTest.java]: Several star notations.

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlPointBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape.wires.handlers.impl;
+
+import com.ait.lienzo.client.core.animation.IAnimationHandle;
+import com.ait.lienzo.client.core.shape.*;
```
Comment (hasys) [inline src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlPointBuilderTest.java]: Star notations.

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlPointBuilderTest.java [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+    private Layer                             layer;
+
+    private Layer                             overLayer;
+
+    private Transform                         transform;
+
+    private WiresConnector                    connector;
+
+    private IDirectionalMultiPointShape       line;
```
Comment (hasys) [inline src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlPointBuilderTest.java]: `transform`, `connector` and `line` can be local variables.

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java [hunk 4]
```diff
@@ -15,30 +15,17 @@
...
-import java.util.HashSet;
-import java.util.Map;
-
-import com.ait.lienzo.client.core.event.NodeDragEndEvent;
-import com.ait.lienzo.client.core.event.NodeDragMoveEvent;
-import com.ait.lienzo.client.core.event.NodeDragStartEvent;
-import com.ait.lienzo.client.core.event.NodeMouseClickEvent;
-import com.ait.lienzo.client.core.event.NodeMouseMoveEvent;
-import com.ait.lienzo.client.core.shape.Circle;
+import com.ait.lienzo.client.core.event.*;
```
Comment (hasys) [inline src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java]: Star notation.

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java [hunk 5]
```diff
@@ -49,81 +36,88 @@
...
     @Mock
-    private SelectionManager selectionManager;
+    private SelectionManager                  selectionManager;
 
     @Mock
-    private SelectionManager.SelectedItems selectedItems;
+    private SelectionManager.SelectedItems    selectedItems;
 
-    private BoundingBox boundingBox;
+    private BoundingBox                       boundingBox;
```
Comment (hasys) [inline src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java]: Can be a local variable.

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java [hunk 6]
```diff
@@ -49,81 +36,88 @@
...
 
     @Mock
-    private WiresLayer wiresLayer;
-
-    private Shape transientPoint;
+    private WiresLayer                        wiresLayer;
 
     @Captor
-    private ArgumentCaptor<Consumer> createConsumer;
+    private ArgumentCaptor<Consumer>          createConsumer;
```
Comment (hasys) [inline src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java]: This field is never used.

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java [hunk 7]
```diff
@@ -49,81 +36,88 @@
...
 
     @Mock
-    private NodeMouseMoveEvent moveEvent;
+    private NodeMouseMoveEvent                moveEvent;
+
+    private WiresConnectorHandlerImpl         tested;
+
+    private Point2DArray                      linePoints;
```
Comment (hasys) [inline src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java]: `linePoints` can be a local variable.

File: src/test/java/com/ait/lienzo/client/core/shape/wires/DefaultSelectionListenerTest.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Mock
+    private WiresConnector                 selectedConnector;
+
+    @Mock
+    private WiresConnector                 deselectedConnector;
+
+    private DefaultSelectionListener       tested;
+
+    private Layer                          layer;
```
Comment (hasys) [inline src/test/java/com/ait/lienzo/client/core/shape/wires/DefaultSelectionListenerTest.java]: Can be a local variable.

File: src/test/java/com/ait/lienzo/client/core/shape/wires/WiresContainerTest.java [hunk 9]
```diff
@@ -18,22 +18,14 @@
...
 import com.ait.lienzo.client.core.shape.MultiPath;
-import com.ait.lienzo.client.core.shape.wires.event.WiresDragEndEvent;
-import com.ait.lienzo.client.core.shape.wires.event.WiresDragEndHandler;
-import com.ait.lienzo.client.core.shape.wires.event.WiresDragMoveEvent;
-import com.ait.lienzo.client.core.shape.wires.event.WiresDragMoveHandler;
-import com.ait.lienzo.client.core.shape.wires.event.WiresDragStartEvent;
-import com.ait.lienzo.client.core.shape.wires.event.WiresDragStartHandler;
-import com.ait.lienzo.client.core.shape.wires.event.WiresMoveEvent;
-import com.ait.lienzo.client.core.shape.wires.event.WiresMoveHandler;
+import com.ait.lienzo.client.core.shape.wires.event.*;
```
Comment (hasys) [inline src/test/java/com/ait/lienzo/client/core/shape/wires/WiresContainerTest.java]: Star notation.

File: src/test/java/com/ait/lienzo/client/core/shape/wires/WiresManagerTest.java [hunk 10]
```diff
@@ -19,12 +19,7 @@
 package com.ait.lienzo.client.core.shape.wires;
 
 import com.ait.lienzo.client.core.Context2D;
-import com.ait.lienzo.client.core.shape.AbstractDirectionalMultiPointShape;
-import com.ait.lienzo.client.core.shape.Attributes;
-import com.ait.lienzo.client.core.shape.Group;
-import com.ait.lienzo.client.core.shape.Layer;
-import com.ait.lienzo.client.core.shape.MultiPath;
-import com.ait.lienzo.client.core.shape.Viewport;
+import com.ait.lienzo.client.core.shape.*;
```
Comment (hasys) [inline src/test/java/com/ait/lienzo/client/core/shape/wires/WiresManagerTest.java]: Star notation.

File: src/test/java/com/ait/lienzo/client/core/shape/wires/WiresShapeTest.java [hunk 11]
```diff
@@ -26,12 +26,7 @@
 import com.ait.lienzo.client.core.shape.IPrimitive;
 import com.ait.lienzo.client.core.shape.MultiPath;
 import com.ait.lienzo.client.core.shape.Rectangle;
-import com.ait.lienzo.client.core.shape.wires.event.WiresResizeEndEvent;
-import com.ait.lienzo.client.core.shape.wires.event.WiresResizeEndHandler;
-import com.ait.lienzo.client.core.shape.wires.event.WiresResizeStartEvent;
-import com.ait.lienzo.client.core.shape.wires.event.WiresResizeStartHandler;
-import com.ait.lienzo.client.core.shape.wires.event.WiresResizeStepEvent;
-import com.ait.lienzo.client.core.shape.wires.event.WiresResizeStepHandler;
+import com.ait.lienzo.client.core.shape.wires.event.*;
```
Comment (hasys) [inline src/test/java/com/ait/lienzo/client/core/shape/wires/WiresShapeTest.java]: Star notation.

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlImplTest.java (hunk 1 above)
Comment (romartin) [reply src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlImplTest.java]: fixed

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlPointBuilderTest.java (hunk 2 above)
Comment (romartin) [reply src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlPointBuilderTest.java]: fixed

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlPointBuilderTest.java (hunk 3 above)
Comment (romartin) [reply src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlPointBuilderTest.java]: fixed

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java (hunk 4 above)
Comment (romartin) [reply src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java]: fixed

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java (hunk 5 above)
Comment (romartin) [reply src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java]: fixed

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java (hunk 6 above)
Comment (romartin) [reply src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java]: fixed

File: src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java (hunk 7 above)
Comment (romartin) [reply src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorHandlerImplTest.java]: fixed

File: src/test/java/com/ait/lienzo/client/core/shape/wires/DefaultSelectionListenerTest.java (hunk 8 above)
Comment (romartin) [reply src/test/java/com/ait/lienzo/client/core/shape/wires/DefaultSelectionListenerTest.java]: fixed

File: src/test/java/com/ait/lienzo/client/core/shape/wires/WiresContainerTest.java (hunk 9 above)
Comment (romartin) [reply src/test/java/com/ait/lienzo/client/core/shape/wires/WiresContainerTest.java]: fixed

File: src/test/java/com/ait/lienzo/client/core/shape/wires/WiresManagerTest.java (hunk 10 above)
Comment (romartin) [reply src/test/java/com/ait/lienzo/client/core/shape/wires/WiresManagerTest.java]: fixed

File: src/test/java/com/ait/lienzo/client/core/shape/wires/WiresShapeTest.java (hunk 11 above)
Comment (romartin) [reply src/test/java/com/ait/lienzo/client/core/shape/wires/WiresShapeTest.java]: fixed

**Comment by romartin** [comment]
@hasys added new commit with the fixes - thanks!

**Comment by romartin** [comment]
rebased and squashed

**Pull request merged**